src/components/marker-chart/Canvas.js

```diff
@@ -114,9 +118,27 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
           ? undefined
           : markerIndexToTimingRow.get(hoveredItem);
 
-      if (newRow !== undefined) {
+      if (newRow !== undefined && newRow !== rightClickedRow) {
+        /**
+         * To avoid multiple renders on the same row rectangle in canvas, first the row
+         * is cleared and then redrawn on using the drawMarkers()
+         **/
+        ctx.clearRect(
+          0, // To include the labels also
+          newRow * rowHeight - viewportTop,
+          containerWidth,
+          rowHeight - 1 // Subtract 1 for borders.
+        );
         this.clearRow(ctx, newRow);
         this.drawMarkers(ctx, hoveredItem, newRow, newRow + 1);
+        // To highlight the row when mouse hovered over
+        ctx.fillStyle = 'rgba(40, 122, 169, 0.2)';
+        ctx.fillRect(
+          0, // To include the labels also
+          newRow * rowHeight - viewportTop,
+          containerWidth,
+          rowHeight - 1 // Subtract 1 for borders.
+        );
```

Comment:
I think it would be valuable to have a local method `highlightRow` taking `(ctx, rowItem)` as parameters. That would make this code more readable.

Reply:
Yes, I agree with this one. I was hesitant about making any extra methods, but I guess we could do this and maybe make one for right-click as well which can be reused if you would like a highlight on double click on a row? (different bc transparency value in rgba are different)

Reply:
Sorry, I wasn't clear on this: I think the highlight should be the same for rightclick vs hover. That's why I'm suggesting to do a common method and use it in both situations.